src-tauri/src/api.rs

```diff
@@ -11,17 +13,41 @@ impl<T> TauriResult<T> for Result<T, Error> {
     }
 }
 
+impl<T> TauriResult<T> for std::io::Result<T> {
+    fn serialize_err(self) -> Result<T, String> {
+        self.map_err(|e| e.to_string())
+    }
+}
+
 #[tauri::command]
 pub async fn get_repositories_view_model() -> Result<Vec<RepositoryViewModel>, String> {
     viewmodel_api::list_repositories().await.serialize_err()
 }
 
 #[tauri::command]
 pub async fn add_repository(url: Url) -> Result<RepositoryViewModel, String> {
-    let (repo, file_path) = viewmodel_api::download_resource_to_file(url)
+    let (repo, file_path) = viewmodel_api::download_resource_to_file(url, false)
         .await
         .serialize_err()?;
     RepositoryViewModel::from_resource(repo, file_path)
         .await
         .serialize_err()
 }
+
+#[tauri::command]
+pub async fn delete_resource(path: PathBuf) -> Result<(), String> {
+    tokio::fs::remove_file(path).await.serialize_err()
```

Comment:
We should check whether we actually own the file first. Add an API method under viewmodel-api